README.markdown

```diff
@@ -456,6 +456,8 @@ Generated code will be placed in the Gradle build directory.
 
 - With `--ts_proto_opt=outputSchema=true`, meta typings will be generated that can later be used in other code generators. If outputSchema is instead specified to be `no-file-descriptor` then we do not include the file descriptor in the generated schema. This is useful if you are trying to minimize the size of the generated schema.
 
+- With `--ts_proto_opt=outputSchemaAsConst=true`, the `protoMetadata` export of `outputSchema=true` is declared using `as const` and `satisfies` (supported by TypeScript since [4.9](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator)), providing more detailed type information.
```

Comment:
Hi @cmd-johnson ! Wdyt of making `outputSchema` a tristate, like `outputSchema=const` would enable this behavior, vs. a net-new flag? We've done that with a few other flags and it's seemed nice so far.

Reply:
That's what I wanted to do at first, but `outputSchema` already can be `true` or `no-file-descriptor`, which would mean that adding an `as const` option would also require adding a `no-file-descriptor-as-const` option or make the option's type an array of a new `OutputSchema` enum (similar to how `outputServices` works) and I didn't want to update all usages of `no-file-descriptor` :sweat_smile:.

If you'd rather go that route than adding another parameter, I can make those changes. I'd prefer going the enum array route in that case, because it scales better if more options get added to the `outputSchema` flag at some point in the future.

Reply:
Ah yep! I was wondering if we'd run into that.

The enum array of `outputSchema=no-file-descriptor,const,etc` approach would be great, since you don't mind; thanks!

Reply:
Done!
I've made it so you can combine the flags by specifying it multiple times, which is the way the `outputServices` flag works:
```
--ts_proto_opt=outputSchema=const,outputSchema=no-file-descriptor
```